### DISCUSSION TITLE
Env's not loading from the next config during tests

### DISCUSSION THREAD

**Discussion by randallmlough** [vercel/next.js#36180] [Help]
This may be by design, but I can't get env's that are added to the next config to load/be accessible during tests. I tried to even use your [jest example](https://github.com/vercel/next.js/tree/canary/examples/with-jest), but I get the same results. I was under the assumption that `next.config.js` is loaded during tests since 12+. Maybe I'm wrong here or maybe env's are intentionally left out. 

My use case is I load a configuration file ( `app.${ENV}.js`) rather than having everything in an `.env` file, similar to [your example](https://github.com/vercel/next.js/blob/canary/examples/with-env-from-next-config-js/next.config.js). But these values never get loaded during testing enviroment.

Brief example
```js
/**
 * @type {import('next').NextConfig}
 */
 const nextConfig = {
    env: {
        test: 'foo'
        ...getConfig(`app.config.js`),
        ...getConfig(`app.${process.env.NODE_ENV}.config.js`),
    }
  }
  
module.exports = nextConfig
```

Is this a limitation by design or am I doing something wrong?

**Comment by icyJoseph** [comment]
In order to gain access to the auto `.env` loading you need to:

```js
const nextJest = require('next/jest')

const createJestConfig = nextJest({
  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
  dir: './',
})

/**
 * @type {import('next').NextConfig}
 */
 const nextConfig = {
    env: {
        test: 'foo'
        ...getConfig(`app.config.js`),
        ...getConfig(`app.${process.env.NODE_ENV}.config.js`),
    }
  }
  
module.exports = createJestConfig(nextConfig)
```

IIRC this is a Next 12 feature, by opting into the SWC compiler: https://nextjs.org/docs/testing#setting-up-jest-with-the-rust-compiler.

I always use a `.env.test` (prior to the feature anyway), I'll give it a go as well.

## Edit

This seems to work only when loading from `.env` files:

> - Loading .env (and all variants) into process.env
